doozer/doozerlib/backend/rebaser.py

```diff
@@ -0,0 +1,1549 @@
+                next_word = word
+
+            # note: there are a number of ways to defeat this logic, for instance by
+            # wrapping commands/args in quotes, or with commands that aren't valid
+            # to begin with. let's not worry about that; it need not be invulnerable.
+
+        if docker_cmd_options:
+            cmd = " ".join(docker_cmd_options) + " " + cmd
+        return changed, cmd
+
+    def _write_osbs_image_config(self, metadata: ImageMetadata, dest_dir: Path, source: Optional[SourceResolution], version: str):
+        # Writes OSBS image config (container.yaml).
+        # For more info about the format, see https://osbs.readthedocs.io/en/latest/users.html#image-configuration.
+
+        self._logger.info('Generating container.yaml')
+        container_config = self._generate_osbs_image_config(metadata, dest_dir, source, version)
+
+        if 'compose' in container_config:
+            self._logger.info("Rebasing with ODCS support")
+        else:
+            self._logger.info("Rebasing without ODCS support")
+
+        # generate yaml data with header
+        content_yml = yaml.safe_dump(container_config, default_flow_style=False)
+        with dest_dir.joinpath('container.yaml').open('w', encoding="utf-8") as rc:
+            rc.write(CONTAINER_YAML_HEADER + content_yml)
+
+    def _generate_osbs_image_config(self, metadata: ImageMetadata, dest_dir: Path, source: Optional[SourceResolution], version: str) -> Dict:
+        """
+        Generates OSBS image config (container.yaml).
+        Returns a dict for the config.
+
+        Example in image yml file:
+        odcs:
+            packages:
+                mode: auto (default) | manual
+                # auto - If container.yaml with packages is given from source, use them.
+                #        Otherwise all packages with from the Koji build tag will be included.
+                # manual - only use list below
+                list:
+                  - package1
+                  - package2
+        arches: # Optional list of image specific arches. If given, it must be a subset of group arches.
+          - x86_64
+          - s390x
+        container_yaml: ... # verbatim container.yaml content (see https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/odcs_integration_with_osbs)
+        """
+
+        # list of platform (architecture) names to build this image for
+        arches = metadata.get_arches()
+
+        # override image config with this dict
+        config_overrides = {}
+        if metadata.config.container_yaml is not Missing:
+            config_overrides = copy.deepcopy(cast(Dict, metadata.config.container_yaml.primitive()))
+
+        # Cachito will be configured if `cachito.enabled` is True in image metadata
+        # or `cachito.enabled` is True in group config.
+        # https://osbs.readthedocs.io/en/latest/users.html#remote-sources
+        cachito_enabled = False
```

Comment:
curious how having this cachito bit of code is going to effect konflux? no effect?

Reply:
I don't think Cachito actually works for Konflux build. I currently keep this section untouched since it doesn't cause any side effects.